src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+    } else {
+        if (this->eyeTimer > 3) {
+            this->eyeTimer--;
+        } else {
+            if (this->eyeTimer != 0) {
+                this->eyeTexIndex = EN_BAL_EYETEX_CLOSED;
+                this->eyeTimer--;
+            } else {
+                this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+                this->eyeTimer = 60;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62ED0.s")
+void EnBal_UpdateCollision(EnBal* this, PlayState* play) {
+    if ((this->actionFunc == EnBal_FloatIdle) || ((this->actionFunc == EnBal_FloatUpToSky) && (this->timer > 50))) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A62FAC.s")
+void EnBal_CheckBalloonPopped(EnBal* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->picto.actor.world.pos.x,
+                    this->picto.actor.world.pos.y + 100.0f, this->picto.actor.world.pos.z, 0xFF, 0xFF, 0xC8, 2);
+        this->collider.base.acFlags &= ~AC_HIT;
+        EnBal_SetupPopBalloon(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A63158.s")
+void EnBal_Update(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A634B4.s")
+    this->actionFunc(this, play);
+    EnBal_CheckBalloonPopped(this, play);
+    EnBal_UpdateCollision(this, play);
+    SkelAnime_Update(&this->skelAnime);
+    if (this->actionFunc != EnBal_FloatIdle) {
+        Actor_MoveWithGravity(&this->picto.actor);
+        Actor_UpdateBgCheckInfo(play, &this->picto.actor, 32.0f, 30.0f, 60.0f,
+                                UPDBGCHECKINFO_FLAG_2 | UPDBGCHECKINFO_FLAG_4);
+    }
+    EnBal_UpdateHead(this, play);
+    if ((this->actionFunc == EnBal_Talk) && (this->prevTextId != 0x1D10)) {
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+        this->picto.actor.world.rot.y = this->picto.actor.shape.rot.y;
+    }
+    EnBal_UpdateShadow(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A634C8.s")
+s32 EnBal_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnBal* this = THIS;
+    Vec3s balloonRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A635DC.s")
+    if (limbIndex == TINGLE_LIMB_BALLOON) {
+        if ((this->prevBalloonAction == EN_BAL_BALACT_NONE) || (this->prevBalloonAction == EN_BAL_BALACT_INFLATE) ||
+            (this->prevBalloonAction == EN_BAL_BALACT_RISE)) {
+            balloonRot.x = Math_SinS(this->timer) * 3640.0f;
+            balloonRot.z = Math_CosS(this->timer) * 3640.0f;
```

Comment:
```suggestion
            balloonRot.x = Math_SinS(this->timer) * 0xE38;
            balloonRot.z = Math_CosS(this->timer) * 0xE38;
```
Can you get away with hex?